api/openapi-spec/swagger.json

```diff
@@ -7983,7 +8099,12 @@
           "description": "NodeAffinity defines constraints that limit what nodes this volume can be accessed from. This field influences the scheduling of pods that use this volume."
         },
         "persistentVolumeReclaimPolicy": {
-          "description": "What happens to a persistent volume when released from its claim. Valid options are Retain (default for manually created PersistentVolumes), Delete (default for dynamically provisioned PersistentVolumes), and Recycle (deprecated). Recycle must be supported by the volume plugin underlying this PersistentVolume. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#reclaiming",
+          "description": "What happens to a persistent volume when released from its claim. Valid options are Retain (default for manually created PersistentVolumes), Delete (default for dynamically provisioned PersistentVolumes), and Recycle (deprecated). Recycle must be supported by the volume plugin underlying this PersistentVolume. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#reclaiming\n\nPossible enum values:\n - `Delete`: means the volume will be deleted from Kubernetes on release from its claim. The volume plugin must support Deletion.\n - `Recycle`: means the volume will be recycled back into the pool of unbound persistent volumes on release from its claim. The volume plugin must support Recycling.\n - `Retain`: means the volume will be left in its current phase (Released) for manual reclamation by the administrator. The default policy is Retain.",
```

Comment:
The language used in these values is all sorts of inconsistent.  Is there a plan to a) document a convention; and b) convert all such comments to that convention?

Reply:
The plan is not there, yet. This feature is not designed to include some mechanism to normalize the comments through our APIs. To set a convention of comments would be out of scope of OpenAPI or any of its features.

Reply:
We can think about a smarter algorithm of description generation, possibly during alpha-to-beta graduation. The current algorithm in `kube-openapi` was the one worked best with the enums defined in `k8s.io/api/core/v1`

Reply:
I will amend the KEP to add a design for a normalized form of comments for enum values and a "linter" that detect malformed enum comments. I will make this part of alpha-to-beta graduation. Would you agree with my plan?

Reply:
Sounds great!  

What does "alpha" mean for this?  It doesn't seem like something that can be gate-disabled?

Reply:
This PR implements a feature that is going alpha in 1.23. It is gated by `OpenAPIEnum` feature gate. The way we implement it is to add the marker to our types, generate OpenAPI spec with enums fields and descriptions (that's why this PR upgrade the `k/kube-openapi` package), and, at run-time, remove enums from the generated spec if the feature gate is disabled.

So basically, the spec served at `/openapi/v2` won't have the new enums fields and description, just like what we have before this PR, unless this feature gate is enabled.

Ref: [KEP] https://github.com/kubernetes/enhancements/issues/2887

Reply:
`swagger.json` is generated with `AllAlpha=true` feature gate. That's why we can see the enums in the OpenAPI spec.

Reply:
Thanks